hack/deploy-karmada.sh

```diff
@@ -118,20 +121,32 @@ kubectl apply -f "${REPO_ROOT}/artifacts/deploy/karmada-etcd.yaml"
 # Wait for karmada-etcd to come up before launching the rest of the components.
 util::wait_pod_ready "${ETCD_POD_LABEL}" "${KARMADA_SYSTEM_NAMESPACE}"
 
+
+# If it provided a karmada API Server IP we can access karmada API Server (cluster by kind), we will create a ClusterIP Service
+# Or we need to create a LoadBalancer service so that we can access karmada API Server outside the karmada-host cluster
+KARMADA_APISERVER_SERVICE_TYPE="ClusterIP"
+if [[ -z "${KARMADA_APISERVER_IP}" ]]; then
```

Comment:
@lfbear If user want to install karmada into a cluster without loadbalance, they need to deploy the service of karmada apiserver first to get the clusterIP?

Reply:
I assume that it only uses `ClusterIP` in `Kind` mode, i.e. developing or testing when the control plane cluster and members clusters are created by Kind. In other cases, such as in production, I think that it is necessary to reveal the Karmda API Server IP out from the control plane cluster. So I guess it can not work correctly to install karmada without loadbalance. Of course, there are several ways to reveal the Karmda API Server IP, I chose the `loadbalance` for the time being

Reply:
I agree to use loadbalance in production. But we should disscuss more about if we need to cover the scenario that user want to install karmada into the cluster without loadbalance and not created by kind.  cc @RainbowMango

Reply:
At least, we should tell them how to modify the script. Or we can add a param, such as useClusterIP. Then, if KARMADA_APISERVER is empty && useClusterIP is true, we still use clusterIP. The default value of useClusterIP can be false.

Reply:
agree +1, we should collect more scenarios, I also wonder the more usage. 
So far, I added parameters one after another, the installation will change to difficult if we can't make the scenario clear. I want to make it easy to start up all the time.